### ISSUE TITLE
VulnerabilityReports are no longer created

### ISSUE THREAD

**Issue by c-dst** [aquasecurity/trivy-operator#1194]
**What steps did you take and what happened:**

Since I ugraded Microk8s to snap 1.27.0/stable on Ubuntu 22.04, images vuln repport don't create anymore objects vulnerabilityreports.aquasecurity.github.io.

The scan job are done, the Pod output the right json repport.

```
{
  "SchemaVersion": 2,
  "ArtifactName": "quay.io/jetstack/cert-manager-controller:v1.11.1",
  "ArtifactType": "container_image",
  "Metadata": {
    "OS": {
      "Family": "debian",
      "Name": "11.6"
    },
    "ImageID": "sha256:b23ed1d12779ca6b5a853d022d363e2edefc2107890342f50eba7f73da6eaa99",
    "DiffIDs": [
      "sha256:d0157aa0c95a4cae128dab97d699b2f303c8bea46914dc4a40722411f50bb40e",
      "sha256:6fbdf253bbc2490dcfede5bdb58ca0db63ee8aff565f6ea9f918f3bce9e2d5aa",
      "sha256:399826b51fcf6c959b7a7e86b89ac1ee6685d64da54e5223e1d182c491a1bbd6",
      "sha256:ff5700ec54186528cbae40f54c24b1a34fb7c01527beaa1232868c16e2353f52",
      "sha256:d52f02c6501c9c4410568f0bf6ff30d30d8290f57794c308fe36ea78393afac2",
      "sha256:e624a5370eca2b8266e74d179326e2a8767d361db14d13edd9fb57e408731784",
      "sha256:1a73b54f556b477f0a8b939d13c504a3b4f4db71f7a09c63afbc10acb3de5849",
      "sha256:d2d7ec0f6756eb51cf1602c6f8ac4dd811d3d052661142e0110357bf0b581457",
      "sha256:4cb10dd2545bd173858450b80853b850e49608260f1a0789e0d0b39edf12f500",
      "sha256:b75a9cf43ee540efc11274b4c0dcd34dc58fff375d16309422aa3c86b3cd2c08",
      "sha256:c5a0e588718a1766cc97b773bda7db1351b076dfcb11d7a478cd6338a28088f0",
      "sha256:d2eabf58d6e7f4f0ec4fd2429dcdabf04199f4e2593d9355e096866f2fe83677"
    ],
    "RepoTags": [
      "quay.io/jetstack/cert-manager-controller:v1.11.1"
    ],
    "RepoDigests": [
      "quay.io/jetstack/cert-manager-controller@sha256:4f1782c8316f34aae6b9ab823c3e6b7e6e4d92ec5dac21de6a17c3da44c364f1"
    ],
  },
...
  "Results": [
    {
      "Target": "quay.io/jetstack/cert-manager-controller:v1.11.1 (debian 11.6)",
      "Class": "os-pkgs",
      "Type": "debian"
    },
    {
      "Target": "app/cmd/controller/controller",
      "Class": "lang-pkgs",
      "Type": "gobinary",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2020-8911",
          "PkgName": "github.com/aws/aws-sdk-go",
          "InstalledVersion": "v1.44.105",
          "Layer": {
            "Digest": "sha256:0a0ad7f5d4ca8df212a06ee2a25eacfd9c66e6a3eecf668699c6760a68f45b56",
            "DiffID": "sha256:b75a9cf43ee540efc11274b4c0dcd34dc58fff375d16309422aa3c86b3cd2c08"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2020-8911",
          "DataSource": {
            "ID": "go-vulndb",
            "Name": "The Go Vulnerability Database",
            "URL": "https://github.com/golang/vulndb"
          },
          "Title": "aws/aws-sdk-go: CBC padding oracle issue in AWS S3 Crypto SDK for golang",


          "Description": "A padding oracle vulnerability exists in the AWS S3 Crypto SDK for GoLang versions prior to V2. The SDK allows users to encrypt files with AES-CBC without computing a Message Authentication Code (MAC), which then allows an attacker who has write access to the target's S3 bucket and can observe whether or not an endpoint with access to the key can decrypt a file, they can reconstruct the plaintext with (on average) 128*length (plaintext) queries to the endpoint, by exploiting CBC's ability to manipulate the bytes of the next block and PKCS5 padding errors. It is recommended to update your SDK to V2 or later, and re-encrypt your files.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-327"
          ],
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:N/S:C/C:H/I:N/A:N",
              "V3Score": 5.6
            },
            "nvd": {
              "V2Vector": "AV:L/AC:L/Au:N/C:P/I:N/A:N",
              "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:N/S:C/C:H/I:N/A:N",
              "V2Score": 2.1,
              "V3Score": 5.6
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:N/S:C/C:H/I:N/A:N",
              "V3Score": 5.6
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2020-8911",
            "https://aws.amazon.com/blogs/developer/updates-to-the-amazon-s3-encryption-client/?s=09",
            "https://bugzilla.redhat.com/show_bug.cgi?id=1869800",
            "https://github.com/advisories/GHSA-f5pg-7wfw-84q9",
            "https://github.com/aws/aws-sdk-go/commit/1e84382fa1c0086362b5a4b68e068d4f8518d40e",
            "https://github.com/aws/aws-sdk-go/commit/ae9b9fd92af132cfd8d879809d8611825ba135f4",
            "https://github.com/aws/aws-sdk-go/pull/3403",
            "https://github.com/google/security-research/security/advisories/GHSA-f5pg-7wfw-84q9",
            "https://github.com/sophieschmieg/exploits/tree/master/aws_s3_crypto_poc",
            "https://nvd.nist.gov/vuln/detail/CVE-2020-8911",
            "https://pkg.go.dev/vuln/GO-2022-0646",
            "https://www.cve.org/CVERecord?id=CVE-2020-8911"
          ],
          "PublishedDate": "2020-08-11T20:15:00Z",
          "LastModifiedDate": "2020-08-18T13:37:00Z"
        },
...

  ]
}
```

**What did you expect to happen:**

The vuln repports where create, like other reports which are always done.

**Anything else you would like to add:**

I try to downgrade the operator to version 0.11.0 or 0.12.0, the problem is the same. 

**Environment:**

- Trivy-Operator version (use `trivy-operator version`): 0.13.2
- Kubernetes version (use `kubectl version`): 1.27.0
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): Ubuntu 22.04 snap mikrok8s 1.27.0/stable

**Comment by xerxist** [comment]
Same on Talos 1.4.1 Kubernetes 1.27.1

**Comment by chen-keinan** [comment]
@c-dst @xerxist any error on `trivy-operator` or scan-job stuck in `completed` status ?

**Comment by xerxist** [comment]
I've since reverted back to 1.26.4 and the problem is gone. Seems to be an issue from 1.27.0 and onward.

**Comment by c-dst** [comment]
> @c-dst @xerxist any error on `trivy-operator` or scan-job stuck in `completed` status ? do you mind sharing the operator `configmaps` ?

scan-job completes normally, trivy-operator and trivy server pod doesn't report errors in logs even with debug enabled.

Most values are defaults. 

``` yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: trivy-operator
  namespace: trivy-operator
  labels:
    app.kubernetes.io/instance: trivy-operator
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: trivy-operator
    app.kubernetes.io/version: 0.13.2
    helm.sh/chart: trivy-operator-0.13.2
  annotations:
    meta.helm.sh/release-name: trivy-operator
    meta.helm.sh/release-namespace: trivy-operator
data:
  compliance.failEntriesLimit: '10'
  configAuditReports.scanner: Trivy
  node.collector.imageRef: ghcr.io/aquasecurity/node-collector:0.0.6
  nodeCollector.volumeMounts: >-
    [{"mountPath":"/var/lib/etcd","name":"var-lib-etcd","readOnly":true},{"mountPath":"/var/lib/kubelet","name":"var-lib-kubelet","readOnly":true},{"mountPath":"/var/lib/kube-scheduler","name":"var-lib-kube-scheduler","readOnly":true},{"mountPath":"/var/lib/kube-controller-manager","name":"var-lib-kube-controller-manager","readOnly":true},{"mountPath":"/etc/systemd","name":"etc-systemd","readOnly":true},{"mountPath":"/lib/systemd/","name":"lib-systemd","readOnly":true},{"mountPath":"/etc/kubernetes","name":"etc-kubernetes","readOnly":true},{"mountPath":"/etc/cni/net.d/","name":"etc-cni-netd","readOnly":true}]
  nodeCollector.volumes: >-
    [{"hostPath":{"path":"/var/lib/etcd"},"name":"var-lib-etcd"},{"hostPath":{"path":"/var/lib/kubelet"},"name":"var-lib-kubelet"},{"hostPath":{"path":"/var/lib/kube-scheduler"},"name":"var-lib-kube-scheduler"},{"hostPath":{"path":"/var/lib/kube-controller-manager"},"name":"var-lib-kube-controller-manager"},{"hostPath":{"path":"/etc/systemd"},"name":"etc-systemd"},{"hostPath":{"path":"/lib/systemd"},"name":"lib-systemd"},{"hostPath":{"path":"/etc/kubernetes"},"name":"etc-kubernetes"},{"hostPath":{"path":"/etc/cni/net.d/"},"name":"etc-cni-netd"}]
  report.recordFailedChecksOnly: 'true'
  scanJob.automountServiceAccountToken: 'true'
  scanJob.podTemplateContainerSecurityContext: >-
    {"allowPrivilegeEscalation":false,"capabilities":{"drop":["ALL"]},"privileged":false,"readOnlyRootFilesystem":true,"runAsUser":0}
  trivy.serverURL: http://trivy-service.trivy-operator:4954
  vulnerabilityReports.scanner: Trivy
binaryData: {}
```

``` yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: trivy-operator-trivy-config
  namespace: trivy-operator
  labels:
    app.kubernetes.io/instance: trivy-operator
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: trivy-operator
    app.kubernetes.io/version: 0.13.2
    helm.sh/chart: trivy-operator-0.13.2
  annotations:
    meta.helm.sh/release-name: trivy-operator
    meta.helm.sh/release-namespace: trivy-operator
data:
  TRIVY_CACHE_DIR: /home/scanner/.cache/trivy
  TRIVY_DB_REPOSITORY: ghcr.io/aquasecurity/trivy-db
  TRIVY_DEBUG: 'true'
  TRIVY_LISTEN: 0.0.0.0:4954
  TRIVY_SKIP_UPDATE: 'false'
  trivy.additionalVulnerabilityReportFields: ''
  trivy.command: image
  trivy.dbRepository: ghcr.io/aquasecurity/trivy-db
  trivy.dbRepositoryInsecure: 'false'
  trivy.javaDbRepository: ghcr.io/aquasecurity/trivy-java-db
  trivy.mode: ClientServer
  trivy.repository: ghcr.io/aquasecurity/trivy
  trivy.resources.limits.cpu: 500m
  trivy.resources.limits.memory: 500M
  trivy.resources.requests.cpu: 100m
  trivy.resources.requests.memory: 100M
  trivy.serverURL: http://trivy-service.trivy-operator:4954
  trivy.severity: UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
  trivy.slow: 'true'
  trivy.supportedConfigAuditKinds: >-
    Workload,Service,Role,ClusterRole,NetworkPolicy,Ingress,LimitRange,ResourceQuota
  trivy.tag: 0.40.0
  trivy.timeout: 5m0s
  trivy.useBuiltinRegoPolicies: 'true'
binaryData: {}
```

**Comment by chen-keinan** [comment]
@c-dst you are also using latest k8s version 1.27.x ?

**Comment by c-dst** [comment]
@chen-keinan Yes, a 1.27.0

**Comment by chen-keinan** [comment]
This propbebly the issue ,we  will support it shortly

**Comment by c-dst** [comment]
I have to stay in this release, and will wait the fix. Thanks.

**Comment by win5923** [comment]
same issue even after installing with Helm, and my k8s version is also 1.27.1

**Comment by chen-keinan** [comment]
> same issue even after installing with Helm, and my k8s version is also 1.27.1

@win5923 what is your `trivy-operator` version ?

**Comment by win5923** [comment]
> > 即使安裝了 Helm 也有同樣的問題，我的 k8s 版本也是 1.27.1
> 
> @win5923`trivy-operator`你的版本是什麼？

@chen-keinan  0.13.2

**Comment by chen-keinan** [comment]
@win5923 it should be fixed with `trivy-operator v0.14.0-rc` can you confirm?

**Comment by win5923** [comment]
I am unable to install `trivy-operator:v0.14.0-rc`. Did I make a mistake in my command? 
```
helm install trivy-operator aqua/trivy-operator \
  --namespace trivy-system \
  --create-namespace \
  --set "trivy.ignoreUnfixed=true" \
  --set "serviceMonitor.enabled=true" \
  --set image.tag="v0.14.0-rc"
```
I  got the following error.
```
Failed to pull image "ghcr.io/aquasecurity/trivy-operator:v0.14.0-rc": rpc error: code = Unknown desc = Error response from daemon: manifest unknown
```

**Comment by chen-keinan** [comment]
> helm install trivy-operator aqua/trivy-operator \
>   --namespace trivy-system \
>   --create-namespace \
>   --set "trivy.ignoreUnfixed=true" \
>   --set "serviceMonitor.enabled=true" \
>   --set image.tag="v0.14.0-rc"

looks ok , maybe indentation or else

**Comment by win5923** [comment]
Successfully installed using the following command.
```
helm install trivy-operator aqua/trivy-operator \
  --namespace trivy-system \
  --create-namespace \
  --set "trivy.ignoreUnfixed=true" \
  --set "serviceMonitor.enabled=true" \
  --set "image.tag=0.14.0-rc"
```
but still can't get VulnerabilityReports
```
kubectl get vulnerabilityreports --all-namespaces -o wide
No resources found
```

**Comment by chen-keinan** [comment]
> > > helm install trivy-operator aqua/trivy-operator
> > > --namespace trivy-system
> > > --create-namespace
> > > --set "trivy.ignoreUnfixed=true"
> > > --set "serviceMonitor.enabled=true"
> > > --set image.tag="v0.14.0-rc"
> > 
> > 
> > looks ok , maybe indentation or else
> 
> Successfully installed using the following command.
> 
> ```
> helm install trivy-operator aqua/trivy-operator \
>   --namespace trivy-system \
>   --create-namespace \
>   --set "trivy.ignoreUnfixed=true" \
>   --set "serviceMonitor.enabled=true" \
>   --set "image.tag=0.14.0-rc"
> ```
> 
> but still can't get VulnerabilityReports
> 
> ```
> kubectl get vulnerabilityreports --all-namespaces -o wide
> No resources found
> ```

thanks for update, I'll check it again (we did upgraded to latest k8s 1.27.x libs) and update you

**Comment by chen-keinan** [comment]
@win5923 found the issue , creating fix now

**Comment by win5923** [comment]
Has the problem been resolved? I updated `trivy-operator` to `0.14.0-rc`, but I still can't see the VulnerabilityReports. Is there any information I need to provide for you?

**Comment by chen-keinan** [comment]
@win5923 a fix has been made but not yet created a release , I'll cut a trivy-operator 0.14.0-rc2 early next week with the release
